src/requests.rs

```diff
@@ -0,0 +1,229 @@
+use std::{thread, time::Duration};
+
+use log::{debug, error, info};
+use rand::rngs::ThreadRng;
+use reqwest::{blocking::Client, header::CONTENT_TYPE};
+
+use crate::{
+    data::{
+        random_gen_dust_concentration, random_gen_humidity, random_gen_pressure,
+        random_gen_temperature, AirPurity, Reading,
+    },
+    error::RequestSchedulerError,
+    URL,
+};
+
+const DEFAULT_REQUEST_AMOUNT: u32 = 1;
+const DEFAULT_TIME_PER_REQUEST: Duration = Duration::from_secs(10);
+const DEFAULT_NUM_THREADS: u32 = 1;
+pub const MAX_NUM_THREADS: u32 = 10;
+
+#[derive(Clone, Copy)]
+pub struct RequestSchedulerBuilder {
+    request_amount: Option<u32>,
+    time_per_request: Option<Duration>,
+    total_time: Option<Duration>,
+    num_threads: Option<u32>,
+}
+
+impl RequestSchedulerBuilder {
```

Comment:
May I suggest that we rename from using the word "Request" to "SensorPost" or something like it. For me, request is a little misleading on the intended functionality here, as it's about posting data to a backend and not requesting data from it. Thoughts?

Reply:
So should I rename it to `SensorPostSchedulerBuilder`? What about all the other mentions of the word `request` in the file?

I feel like "request" does indeed not make too much sense here.

Reply:
Maybe something like `PostSchedulerBuilder` would be adequate unless you think we might have different types of schedulers?